projects/demo/src/app/datepicker/datepicker-with-action-buttons.html

```diff
@@ -0,0 +1,70 @@
+<!--
+  ~ Copyright (c) 2016-2023 VMware, Inc. All Rights Reserved.
+  ~ This software is released under MIT license.
+  ~ The full license information can be found in LICENSE in the root directory of this project.
+  -->
+
+<h2>With Action buttons</h2>
+<form clrForm [formGroup]="dateForm" novalidate>
+  <section class="date-section">
+    <div class="features">
+      <div class="card">
+        <span class="title"> Features </span>
+        <div class="features-list">
+          <clr-toggle-wrapper>
+            <label>Confirmation Action Buttons</label>
+            <input clrToggle type="checkbox" formControlName="showActionButtons" name="showActionButtons" />
+          </clr-toggle-wrapper>
+        </div>
+      </div>
+    </div>
+
+    <div class="flex-section">
+      <div class="card" data-heading="Date Picker">
+        <div class="card-content">
+          <clr-date-container [showActionButtons]="dateForm.get('showActionButtons')?.value">
+            <label for="dateControl">Date</label>
+            <input
+              id="dateControl"
+              type="date"
+              clrDate
+              formControlName="date"
+              [min]="dateForm.get('minDate').value"
+              [max]="dateForm.get('maxDate').value"
+            />
+            <clr-control-error *clrIfError="'min'">
+              <ng-container *ngIf="dateForm.get('date').errors?.min?.min">
+                Minimum date should be {{dateForm.get('date').errors?.min?.min}}
+              </ng-container>
+            </clr-control-error>
+            <clr-control-error *clrIfError="'max'">
+              <ng-container *ngIf="dateForm.get('date').errors?.max?.max">
+                Maximum date should be {{dateForm.get('date').errors?.max?.max}}
+              </ng-container>
+            </clr-control-error>
+          </clr-date-container>
+        </div>
+        <div class="card-validations">
+          <div class="form-input-wrapper clr-form-horizontal">
+            <ng-container
+              *ngTemplateOutlet="validators; context: { minDateFc: 'minDate', maxDateFc: 'maxDate' }"
+            ></ng-container>
+          </div>
+        </div>
+      </div>
+    </div>
+  </section>
+</form>
+<ng-template #validators let-minDateFc="minDateFc" let-maxDateFc="maxDateFc">
+  <ng-container [formGroup]="dateForm">
+    <clr-input-container>
+      <label>Min Date</label>
+      <input type="text" placeholder="yyyy-mm-dd" [formControlName]="minDateFc" clrInput />
+    </clr-input-container>
+
+    <clr-input-container>
+      <label>Max Date</label>
+      <input type="text" placeholder="yyyy-mm-dd" clrInput [formControlName]="maxDateFc" />
+    </clr-input-container>
+  </ng-container>
+</ng-template>
```

Comment:
```suggestion
        <div class="card-validations">
          <div class="form-input-wrapper clr-form-horizontal" [formGroup]="dateForm">
            <clr-input-container>
              <label>Min Date</label>
              <input type="text" placeholder="yyyy-mm-dd" [formControlName]="'minDate'" clrInput />
            </clr-input-container>

            <clr-input-container>
              <label>Max Date</label>
              <input type="text" placeholder="yyyy-mm-dd" clrInput [formControlName]="'maxDate'" />
            </clr-input-container>
          </div>
        </div>
      </div>
    </div>
  </section>
</form>
```

This can be simplified. The `ng-template` is not needed.

Reply:
This can still be cleaned up. Feel free to cherry-pick 4b542afd359df6c8cff098b7df96f00d4a155da0.